torch/csrc/distributed/c10d/DispatchOps.cpp

```diff
@@ -0,0 +1,38 @@
+#include <torch/csrc/distributed/c10d/DispatchOps.h>
+
+#include <aten/src/ATen/core/PythonModeTLS.h>
+#include <torch/library.h>
+
+namespace c10d {
+namespace {
+c10::intrusive_ptr<ProcessGroup::Work> broadcast(const c10::intrusive_ptr<ProcessGroup>& process_group,
+    at::TensorList tensors, int64_t root_rank = 0, int64_t root_tensor = 0, int64_t timeout = -1) {
+  auto tensor_vec = tensors.vec();
+  return process_group->broadcast(tensor_vec,
+      BroadcastOptions {root_rank, root_tensor, std::chrono::milliseconds(timeout)});
+}
+
+TORCH_LIBRARY(c10d, m) {
+  // The following ProcessGroup and Work definations are more like declarations.
+  // They don't expose the details of the two classes into TorchScript.
+  m.class_<ProcessGroup>("ProcessGroup")
+    .def(torch::init<int64_t, int64_t>());
+  m.class_<ProcessGroup::Work>("Work")
+    .def(torch::init<>());
+  // It's important to register the op to the CompositeExplicitAutograd key to enable
+  // __torch_dispatch__.
+  m.def("broadcast", dispatch(c10::DispatchKey::CompositeExplicitAutograd, broadcast));
+}
+}  // namespace
+
+c10::intrusive_ptr<ProcessGroup::Work> dispatch_broadcast(const c10::intrusive_ptr<ProcessGroup>& process_group,
+    at::TensorList tensors, const BroadcastOptions& opts) {
+  auto op = c10::Dispatcher::singleton().findSchemaOrThrow("c10d::broadcast", "")
```

Comment:
Should we cache the returned `op`? Collective ops such as broadcast can be called pretty frequently in model training. Not sure how much price we pay each time we call `findSchemaOrThrow()`.

Reply:
conventionally this is marked static so it's cached. Should be done so here